src/main/java/org/killbill/billing/plugin/vertex/VertexActivator.java

```diff
@@ -22,51 +22,58 @@
 import javax.servlet.Servlet;
 import javax.servlet.http.HttpServlet;
 
+import org.killbill.billing.invoice.plugin.api.InvoicePluginApi;
 import org.killbill.billing.osgi.api.Healthcheck;
 import org.killbill.billing.osgi.api.OSGIPluginProperties;
 import org.killbill.billing.osgi.libs.killbill.KillbillActivatorBase;
-import org.killbill.billing.invoice.plugin.api.InvoicePluginApi;
 import org.killbill.billing.plugin.api.notification.PluginConfigurationEventHandler;
-import org.killbill.billing.plugin.core.config.PluginEnvironmentConfig;
 import org.killbill.billing.plugin.core.resources.jooby.PluginApp;
 import org.killbill.billing.plugin.core.resources.jooby.PluginAppBuilder;
+import org.killbill.billing.plugin.vertex.client.CalculateTaxApi;
+import org.killbill.billing.plugin.vertex.client.TransactionApi;
+import org.killbill.billing.plugin.vertex.oauth.OAuthClient;
 import org.osgi.framework.BundleContext;
 
 public class VertexActivator extends KillbillActivatorBase {
 
     public static final String PLUGIN_NAME = "killbill-vertex";
 
-    private VertexConfigPropertiesConfigurationHandler vertexConfigPropertiesConfigurationHandler;
+    private VertexTransactionApiConfigurationHandler vertexTransactionApiConfigurationHandler;
```

Comment:
I don't think we need separate clients for calculateTaxApi and transactionApi. Authentication token is the same and they are both are actually part of [ Vertex O series API tax calc api](https://tax-calc-api.vertexcloud.com/api/index.html). In Avalara we have another case: there are two separate APIs (free one which is called Rates API) and subscription based one (Avatax). @KonradLubianka What do you think?

Reply:
I added calculator class which for now use only VertexCalculateTaxApiConfigurationHandler. We will see if VertexTransactionApiConfigurationHandler will be needed